packages/mui-material/src/FilledInput/FilledInput.js

```diff
@@ -91,9 +91,16 @@ const FilledInputRoot = styled(InputBaseRoot, {
         // See https://github.com/mui/material-ui/issues/31766
         transform: 'scaleX(1) translateX(0)',
       },
-      [`&.${filledInputClasses.error}:after`]: {
-        borderBottomColor: (theme.vars || theme).palette.error.main,
-        transform: 'scaleX(1)', // error is always underlined in red
+      [`&.${filledInputClasses.error}`]: {
+        [`&:before, &:hover:not(.${filledInputClasses.disabled}):before`]: {
+          borderBottomColor: (theme.vars || theme).palette.error.main,
+        },
```

Comment:
I rather do this instead.

```js
// line 115
[`&:hover:not(.${filledInputClasses.disabled}):not(.${filledInputClasses.error}):before`]: {
  borderBottom: `1px solid ${(theme.vars || theme).palette.text.primary}`,
},
```

Reply:
Good point, this will also fix a difference between MUI & Material on hover bottom-border width. I will push the changes.